### ISSUE TITLE
Problem with storage on a NAS under Windows

### ISSUE THREAD

**Issue by rperr** [SickRage/SickRage#3359]
### Before submitting your issue:

Enable debug logging in SickRage settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit: Latest Update
OS:  Windows 10
What you did: After the update, I clicked into one of my shows and saw that it said the folder was missing and size was 0.00 B even though the Show List shows several GB (I have checked several other shows and all show the same behavior). I tried refreshing, but no luck with that...

What happened: SickRage says the folder is missing and no files exist.  The show folder and files are definitely still there but appears that one of the recent commits may have removed the ability to specify the folder of a show using the '\\SEAGATE-D4\Media\TV Shows' format.  I use a NAS and have used the ''\\SEAGATE-D4\Media\TV Shows' as my root since I installed SickRage several months ago.  I also have the NAS mapped as my Z: drive and tried going into the show settings, but it does not show Z: in the list (think it was always like that which is why I used the \\SEAGATE... format).  If I should do it a different way, I'm more than happy to change the way I specify the show folders...  It also won't postprocess any newly downloaded shows.

What you expected: To be able to properly manage my library and postprocess files to move them to my NAS.

Logs:
```
SHOW REFRESH:
2017-03-07 21:46:33 INFO     SHOWQUEUE-FORCE-UPDATE :: The show dir \\SEAGATE-D4\Media\TV Shows\Lucifer is missing, not bothering to change the episode statuses since it'd probably be invalid

POST-PROCESSING:
2017-03-07 22:12:43 INFO     POSTPROCESSOR-AUTO :: Processing E:\Complete\Beavis.and.Butt.Head.S09E06.mkv (None)
2017-03-07 22:12:43 WARNING  POSTPROCESSOR-AUTO :: A path is required for the destination. Check the root dir and show locations are correct for School Test (I got '\\SEAGATE-D4\Media\TV Shows\Beavis and Butt-Head')
2017-03-07 22:12:43 WARNING  POSTPROCESSOR-AUTO :: Not enough space to continue PP, exiting
2017-03-07 22:12:43 WARNING  POSTPROCESSOR-AUTO :: Processing failed for E:\Complete\Beavis.and.Butt.Head.S09E06.mkv
```

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [x] Post debug logs, either inline (for smaller logs) or using gist
- [x] Please wrap inline logs inside \`\`\` and \`\`\` for readability

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by miigotu** [comment]
Nas asleep? Maybe it was offline? Did you try restarting the NAS and then SickRage?

**Comment by miigotu** [comment]
Have you recently done a reinstall? Make sure the service is using your logon rather than the system account, so that it has the right credentials for the UNC path.

Trying some adjustments anyways.
@sharkykh what do you think about adding a feature for SR to map drives on startup?
The user could specify drive letter, user/pass if required, and UNC path.
Then we map it on startup or change
And the user can select the drive then in the browse dialogs

**Comment by sharkykh** [comment]
Sure, I'll add it to my lists of features.
(Note to self) We should add it next to root dirs in the config.

**Comment by miigotu** [comment]
Only when platform == 'Windows' ofc

**Comment by rperr** [comment]
Have tried restarting both NAS and PC. Also grabbed the latest updates just in case. No luck.

**Comment by miigotu** [comment]
https://github.com/SickRage/SickRage/issues/3359#issuecomment-285451899

**Comment by thatdude123** [comment]
Hey guys, 

I tried that commit, and unfortunately it's a no-go. I also tried creating an symlink from a folder on the c drive to the folder on my nas, and it also didn't work. I've got the nas mapped on a driveletter, none of it works. The service is running under my account, I've got a domain controller setup and it's an account with full permissions on that network share, still doesn't work :/

Any ideas? I'm really not the best in python so figuring this out would take me forever lol. 

Thanks!

**Comment by thatdude123** [comment]
Let me know if you guys want me to try anything. I was going to try and revert back to a previous version, but am weary of going too far back and breaking things lol I'll be online later tonight too.

Cheers.

**Comment by rperr** [comment]
Me too - let me know if I can test anything or help in any way.  I don't know Python or anything but willing to help however I can.  Thinking it may have a hard time with the double slash at the beginning of the UNC path (at least I think Python looks at a double slash as a single slash), so maybe it's not escaping it the right way or something?

**Comment by miigotu** [comment]
Thing is it works for tons of other users, or we would be more convinced =p

Either way, we have some changes in the works that will help eliminate user error 😎

**Comment by thatdude123** [comment]
Hey let me know if you've got anything you're thinking where it could be user error, I'm all ears. This worked fine, then suddently stopped without me touching the machine it's running on lol. I'm also a dev so I know about problems between keyboards and chairs ;)

**Comment by miigotu** [comment]
Either machine using a Windows insider build?

**Comment by thatdude123** [comment]
Running Win 10 pro, version 1607, build 14393.953 
Nope, not insider :)

**Comment by rperr** [comment]
Nope, no insider build. Same OS Win 10 Pro v1607 build 14393.953.  Same experience as thatdude.  It was working just fine then suddenly just didn't - last successful postprocessing was 3/7 at 7:32AM EST (that's the last successful download according to my history so I'm guessing it was something soon after that).

I also used to be a developer so I know what you mean about user error...  Not saying I couldn't do anything wrong, but I don't think I have a pebcak or ID-10-T error over here. ;-)

**Comment by rperr** [comment]
One other thing I just noticed - if I try editing my show and changing the folder to my mapped drive, I get an error telling me the folder doesn't exist.  I do not get that error when I save it with the correct UNC path (not sure if it runs the same check or not on save if I don't change the folder name). However, the show screen still shows the folder as missing.

Went back and looked at the changelog - there was a change to UTF-8 (fix 2292) around that time.  Doing a google search of UTF-8 & \\ led me to this page - http://stackoverflow.com/questions/30587359/how-to-display-convert-a-string-of-utf-8-to-the-proper-symbol

That page seems to indicate that the UTF-8 encode may translate \\ to \ - or maybe its completely unrelated...

**Comment by sharkykh** [comment]
@rperr 
I have Windows 10 on the exact same version, and I don't seem to have any issue connecting to a mapped network folder on my other Windows 7. [I just mapped the UNC Path to a drive.](http://www.laptopmag.com/articles/map-network-drive-windows-10)

It works and acts just like a normal drive, **It also shows up in the file/folder browser on SickRage.**

So it's very weird that it doesn't show up for you...

**Comment by miigotu** [comment]
When run as a service, it starts the service before mapped drives are mounted on startup, so the nest way is to use UNC paths typed right into the browse dialog.

I'm going to add the shares settings later tonight to make this a bit simpler, maybe even autodiscover shares when you click a button... eg: It will scan /24 on your localnetwork for smb shares and list them for you to select from.

**Comment by thatdude123** [comment]
As said on the comit, it now works when giving it the UNC drive, but not using maped drives. I'm cool with that!

Thanks guys, really appreciated.

**Comment by miigotu** [comment]
TBH, I dont know how that fix didnt get merged into develop OR master when I made the fix.... I swear I merged it in -.-

**Comment by thatdude123** [comment]
It's the source control zombies man. Always the zombies :)

**Comment by rperr** [comment]
Still not working for me.  Tried uninstalling and reinstalling. What am I doing wrong?

**Comment by rperr** [comment]
I've tried everything I can think of but it isn't working.  The UNC path works on Plex which is installed on the same PC.  I'm open to any suggestions that anyone has.

**Comment by miigotu** [comment]
Change the service to run as your user.

**Comment by rperr** [comment]
Tried that, but the service won't even start now due to a logon failure, so I switched it back.  it still won't recognize the UNC...

**Comment by miigotu** [comment]
You HAVE to run it as your user, or you CANT use network shares.
Take ownership of C:\SickRage (recursively)
When you change the user on the service it's not just your username. Type the username and then click "Find Now" or something like that on the right side.  Pick the one that has the hostname\username for your user.
You have to have a password set for your user login.
You have to put this password in the service login settings.

**Comment by rperr** [comment]
I guess I'm just confused because it was working fine.  I only have one user on this computer and haven't made any changes to that user (other than anything I've tried to get this working).  I don't have a password set on the account so that it will automatically start everything when it reboots (and so my wife can use the PC if she needs to).  I used the Check Names and it found it (starting the service still didn't work, maybe because of the password thing).

**Comment by miigotu** [comment]
Try setting the share side to allow "Everyone", and restart it

**Comment by rperr** [comment]
Already set that way.  It's a Seagate NAS and the folder is shared with Public.  What is really confusing to me is that I have Plex running on the same PC and it sees the NAS share just fine.

**Comment by miigotu** [comment]
Actually, the firewall is likely the issue.

re-enable the firewall, and then at an admin command prompt:
```bash
net stop sickrage
C:\SickRage\Python\bin\python.exe C:\SickRage\SickRage\SickBeard.py --datadir=C:\SickRage\Data
(firewall request pops up, allow it)
ctrl+c
C:\SickRage\Python\bin\pythonw.exe C:\SickRage\SickRage\SickBeard.py --datadir=C:\SickRage\Data
(firewall request pops up, allow it)
ctrl+c
net start sickrage
```

**Comment by rperr** [comment]
Tried allowing the firewall rule, didn't work.  Tried disabling the firewall, still doesn't work.  Why the heck would it just stop working?  I really like using SR, wish I could figure this out... *sigh*

**Comment by miigotu** [comment]
Can you access the share in a browse dialog in SR like I said by typing it in?

Have you tried using the ip instead of the server hostname?
\\\\192.168.x.x\blah

**Comment by rperr** [comment]
Nope, it won't show up at all, and I tried using the IP with no success. I get <tt> \ \10.0.0.100\Media\TV Shows </tt> does not exist error

**Comment by miigotu** [comment]
Try it in your browser? 
file:///\\\\server\share\

**Comment by rperr** [comment]
Gives me a listing of the folder contents

**Comment by miigotu** [comment]
Run the service as your user.... you still havent accomplished that part.

**Comment by sharkykh** [comment]
By the way, you can still [**_make windows log in to your account automatically._**](https://www.w7forums.com/threads/automatic-login.420/) even with a password set.

**Comment by sharkykh** [comment]
@rperr 
I might have an issue similar to yours but nothing that has to do with SickRage.
Kodi, and Groove Music in fact - Video playback would just stop in Kodi, and music playback would buffer in Groove Music even though I have the share mapped to a drive and it it reading from it.

I don't know yet if it solved this issue for me, but I think you should try it too.
Be sure to restart both of the computers (server and client) too.

> ## Mapped Drive Connection to Network Share May Be Lost
> ### Symptoms
> On a computer that runs one of the versions of Windows that is listed at the beginning of this article, if you map a drive to a network share, the mapped drive may be disconnected after a regular interval of inactivity, and Windows Explorer may display a red "X" on the icon of the mapped drive. However, if you try to access or browse the mapped drive, it reconnects quickly.
> [...]
> To turn off the autodisconnect feature, open a command prompt, type the following line, and then press ENTER:
> `net config server /autodisconnect:-1`

**Comment by rperr** [comment]
Running the service as a user instead of system did the trick!  Not sure why it was working before, but it's fixed now.  I did have to re-add all my shows (for some reason, even though all my shows were listed on the show list SR didn't recognize that the folders were already in the database when I went to add a show I removed when I was trying to make everything work ).  So I'm waiting for 6+ TB of shows to process... lol

Thanks SOOOOO much for all the help!!!

**Comment by miigotu** [comment]
That tells me that your nas is not sharing "Public" to unauthenticated users.
Public is just a library name, it does not mean it is open to the world in all cases.

**Comment by KiwiNibbler** [comment]
Anyone have any other suggestions?  I have mapped my network drives.  But they do not show as valid drives in Sickrage.  So, I used the \\IP_Address\Public\Yadda_Yadda to show where the shows are stored.  It works fine.  But it doesn't work for the post processing folder.  If I use the IP address, I get an error:

POSTPROCESSOR-AUTO :: A path is required for the destination. Check the root dir and show locations are correct for SHOW_NAME (I got '\\192.168.47.104\Public\Yadda_Yadda

I am unable to run the services as my user account.  I know my password.  I even reset it.  But when I change the service to run as my user account instead of the default, I get an error that the credentials are incorrect.  They aren't.  It is probably tied to the stupid Windows 10 thing that passwords are tied to live.com accounts.  It's the only thing I can think of.

I tried using the UNC:  \\Drive_Name\Public\Yadda_Yadd  But Sickrage won'taccept that.

Stuck...

**Comment by miigotu** [comment]
Try starting it with 2 backslashes... `'\\192.168.47.104\Public\Yadda_Yadda`

**Comment by sharkykh** [comment]
@miigotu Markdown escapes the slashes in his post,
if you edit it you can see that he did.

**Comment by miigotu** [comment]
ahh, lol.... I cant/wont test with windows so thats all you haha

**Comment by sharkykh** [comment]
@miigotu #3823

**Comment by miigotu** [comment]
Try disabling windows firewall on both SR machine and share machine for a test

**Comment by sharkykh** [comment]
@miigotu look at the issue I linked. He figured it out

**Comment by miigotu** [comment]
Confusing...

**Comment by KiwiNibbler** [comment]
I thought I figured it out.  It works fine with the IP address for configuring the folder where the shows are saved.  But it doesn't work for post-processing.

I'll try the firewall later, but I'm almost 100% sure that it isn't the issue.  Especially since the PC I am using is on the same network as the NAS.

**Comment by miigotu** [comment]
Same network doesnt mean anything, the firewall doesnt have rules for allowing python.exe and everything is blocked by default.  Since it is ran as a service the firewall does not pop up a request to allow it automatically.

**Comment by KiwiNibbler** [comment]
I checked the firewall rules and it appears that anything Python or Sickrage related is allowed inbound and outbound.  But, to make sure, I turned off the firewall.  It made no difference.

**Comment by KiwiNibbler** [comment]
If it could run as the default service but keep my mapped drives, it would fix the problem.  But that doesn't work.  Well, it works for SABnzbd, but not SR.  Any chance you could figure out how SABnzbd solved the mapping problem and apply it to SR?

**Comment by miigotu** [comment]
pretty sure if you set the service to run as your login user and set it for delayed start mapped drives might work

**Comment by KiwiNibbler** [comment]
It won't run as my user account.  I set it to do so.  But it comes back with a credentials error.  It is part of this glorious Windows 10 thing being tied to your "Live" or "Skype" account.  I know my password.  But the service won't run as me.

**Comment by miigotu** [comment]
https://support.microsoft.com/en-us/instantanswers/5de907f1-f8ba-4fd9-a89d-efd23fee918c/create-a-local-user-or-administrator-account-in-windows-10

Other than that unless you or someone else can track down the issue or how sabnzbd makes it work we can't do anything about it.

**Comment by KiwiNibbler** [comment]
I'm not sure why there is a link on how to create user or admin accounts.  The problem is that the service will not run as my account.  It is a Windows issue that I am unable to resolve.  If I change the service to run as my account, I get an error telling me that the login is incorrect (Error 1069).  I've reset my password and I KNOW it is correct.  The username "checks" ok, but it still doesn't work.  Odd.

As a work around, I just run the application from a command prompt instead of as a service.  I could add it as a startup item I guess.  Strange though.

**Comment by sharkykh** [comment]
@KiwiNibbler Can't actually test this, but have you tried using your email address as the username?

**Comment by KiwiNibbler** [comment]
Oddly enough, yes, just an hour or so ago.  I tried upper, lower, email address, etc.  Very strange.  I'm certain it is a Windows "undocumented feature".

Oddly though, SR doesn't care who the service is run as for the folders for the show locations.  But it does matter for the post processing folders.  I am using my NAS for both.  It can find it (using the IP address) for the show folders, but not for the post-processing.  Maybe I'm special.....  =0)

**Comment by miigotu** [comment]
The link was to suggest creating a second user account, and running SR as that user, then give that user permissions for your files and your user permissions of it's files, and try using mounted drives with the alternate account and also mounting them with your account.

**Comment by miigotu** [comment]
Or you could just set up a media server running linux which is the recommended way for everything involving a media server.

**Comment by KiwiNibbler** [comment]
Ah, thanks @miigotu , that makes sense.  After I posted, I wondered if you were suggesting a second user.  But, I got it working now.

I had it running perfectly on a Raspberry Pi.  It worked for almost two years.  After hundreds of tweaks, it was perfect, running SR, SABnzbd, Couch Potato, and most recently Headphones.  But, apparently I reached the read/write lifespan od my micro-SD card and it died a horrible death.  I was quite disappointed.  I lost all of my perfectly worked configs and post-processing scripts.

So, I just put it all onto my laptop, which never leaves the house.  One day, I'll rebuild the RPi, but I was just a bit too frustrated to do it for now.  (Note to self:  make copies of configs)
